server/channels/api4/websocket.go

```diff
@@ -44,14 +45,15 @@ func connectWebSocket(c *Context, w http.ResponseWriter, r *http.Request) {
 	// We initialize webconn with all the necessary data.
 	// If the queues are empty, they are initialized in the constructor.
 	cfg := &platform.WebConnConfig{
-		WebSocket:     ws,
-		Session:       *c.AppContext.Session(),
-		TFunc:         c.AppContext.T,
-		Locale:        "",
-		Active:        true,
-		PostedAck:     r.URL.Query().Get(postedAckParam) == "true",
-		RemoteAddress: c.AppContext.IPAddress(),
-		XForwardedFor: c.AppContext.XForwardedFor(),
+		WebSocket:        ws,
+		Session:          *c.AppContext.Session(),
+		TFunc:            c.AppContext.T,
+		Locale:           "",
+		Active:           true,
+		PostedAck:        r.URL.Query().Get(postedAckParam) == "true",
+		RemoteAddress:    c.AppContext.IPAddress(),
+		XForwardedFor:    c.AppContext.XForwardedFor(),
+		DisconnectReason: r.URL.Query().Get(disconnectReasonParam),
```

Comment:
We are blindly taking the value of the query param here without any validation, which means this is a threat vector for any user. For example, anybody can write a custom websocket client and send hundreds of different disconnect reasons leading to very high cardinality in our Prometheus instance potentially making a DOS attack.

Let's have a fixed set of reasons, and let this query param go through that layer and extract the correct reason, and "unknown" for anything else.

Reply:
Apologies for the delay - just pushed an update. I'm now only allowing standard websocket error codes (in the range [1000, 1016]), and a custom error code for ping timeouts (4000), and a custom error code for sequence number mismatches (4001). Since these are all numerical error codes, I renamed some of the parameters, fields, and variables to better align with what the data is. Let me know what you think.